From b7b11faa8b3c5eb6081779f87af191ba0cbc4f6e Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Wed, 07 Apr 2021 15:39:54 +0300
Subject: [PATCH] [SW-30456] greco_arc_gnu: spi: add hl_spi_baudr_get

Add hl_spi_baudr_get(), taken from greco_arc. This is because common
code from i2cslave_indirect module is calling spi functionality which
results with link error due to missing function.

Change-Id: I67fc6a831f452e01e94b43cae2d265fcba81de16
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/greco_arc_gnu/asic.c b/greco_arc_gnu/asic.c
index fcc3647..8ac4a73 100644
--- a/greco_arc_gnu/asic.c
+++ b/greco_arc_gnu/asic.c
@@ -75,3 +75,14 @@
 {
 	sram_scrambling_init();
 }
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor;
+	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
+						HL_PLL_DIV3);
+
+	factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
+
+	return factor > 2 ? factor : 2;
+}
